docs/recipes/isolating-mongodb-integration-tests.md

```diff
@@ -0,0 +1,106 @@
+# Setting up AVA for Isolated MongoDB Integration Tests
+
+This recipe outlines how to run disposable MongoDB databases in your AVA tests with per-test isolation.
+This uses `mongomem` and is available on [npm](https://www.npmjs.com/package/mongomem).
+
+`mongomem` is a package that allows you to quickly run a temporary MongoDB server locally.
+It uses temporary file storage which is destroyed when the server stops. 
+
+## Install MongoDB in-memory Server (MongoMem)
+In the root directory of your application, run:
+
+```console
+$ npm install --save-dev mongomem
+```
+
+## Using MongoMem
+In your test file, import the module and run the server.
+ 
+**Make sure to run the server at the start of your file, outside of any test cases.**
+
+```javascript
+import test from 'ava';
+import { MongoDBServer } from 'mongomem';
+
+test.before('start mongodb server', async t => {
+  await MongoDBServer.start();
+})
+
+test('some feature', async t => {
+	const connectionString = await MongoDBServer.getConnectionString();
+	
+	// connectionString === 'mongodb://localhost:27017/3411fd12-b5d6-4860-854c-5bbdb011cb93'
+	// use connectionString to connect to the database with a client of your choice. See below for usage with Mongoose.
```

Comment:
This has tab indentation, unlike the other code samples. It looks a bit odd when the Markdown is rendered.